src/liquid/utils/cloneAttributes.ts

```diff
@@ -1,28 +1,27 @@
-import { JSXBase } from '@stencil/core/internal'
-
-export const cloneAttributes = <B extends HTMLElement>(
-  el: B,
-  attributesToIgnore: string[] = []
-) =>
-  Object.values(el.attributes).reduce<JSXBase.HTMLAttributes<B>>(
-    (acc, { name, value }) => {
-      if (
-        !name ||
-        [
-          'style',
-          'id',
-          'class',
-          'slot',
-          'part',
-          ...attributesToIgnore,
-        ].includes(name)
-      ) {
-        return acc
-      }
+export function cloneAttributes(attributesToIgnore: string[] = []) {
+  const attributes = []
+  for (const key in this) {
+    // Component props are getters. Getters don't have a descriptor.
+    // So we can check for component props as follows:
+    if (!Object.getOwnPropertyDescriptor(this, key) && this[key]) {
+      attributes.push({
+        name: key.replaceAll(/([A-Z])/g, '-$1').toLowerCase(),
+        value: this[key],
+      })
+    }
+  }
+  return Object.values(attributes).reduce((acc, { name, value }) => {
```

Comment:
What happens to other actual attributes, like the `aria-*` ones? Shouldn't we clone them, too?

Reply:
Well, with the change we don't support anything that is not a prop. We could add a set of props for controlling aria related stuff, such as ariaHidden, ariaDisabled etc. but I really don't know if this is a good idea, because there are a lot of them. Can you come up with an example where we would need that feature?

Reply:
Actually, I can think of examples myself. A simple aria-hidden... Anyway, I added another commit which makes sure we also clone attributes which are not defined as props explicitly. These will not be reactive of course.